baseplate/clients/memcache/__init__.py

```diff
@@ -129,9 +130,35 @@ class MemcacheContextFactory(ContextFactory):
 
     """
 
-    def __init__(self, pooled_client: PooledClient):
+    PROM_PREFIX = "bp_memcached_pool"
+    PROM_LABELS = ["pool"]
+
+    promTotalConnections = Gauge(
+        f"{PROM_PREFIX}_size",
+        "Maximum size of this pool",
+        PROM_LABELS,
+    )
+
+    promUsedConnections = Gauge(
+        f"{PROM_PREFIX}_in_use",
+        "Number of connections in this pool currently in use",
+        PROM_LABELS,
+    )
+
+    promFreeConnections = Gauge(
+        f"{PROM_PREFIX}_free",
+        "Number of free connections in this pool",
+        PROM_LABELS,
+    )
+
+    def __init__(self, pooled_client: PooledClient, name: str = "memcache"):
         self.pooled_client = pooled_client
 
+        pool = self.pooled_client.client_pool
+        self.promTotalConnections.labels(name).set_function(lambda: pool.max_size)
+        self.promFreeConnections.labels(name).set_function(lambda: len(pool.free))
+        self.promUsedConnections.labels(name).set_function(lambda: len(pool.used))
+
     def report_memcache_runtime_metrics(self, batch: metrics.Client) -> None:
```

Comment:
i wonder why this memcache method is `report_memcache_runtime_metrics` but all the other clients the equivalent method is `report_runtime_metrics`. I wonder why the difference.

Reply:
Hm, this might be a bug unless it's called outside of the baseplate. It's not referenced anywhere, while those with the latter name are called from https://github.com/reddit/baseplate.py/blob/48bcb0b099608aa8b8eecbf97cb929537d68753d/baseplate/__init__.py#L525

Reply:
Heh https://github.snooguts.net/reddit/reddit-service-post/blob/b0dbc37074a4299fdb916eaddd6b6cb1fe1fd568/post/__init__.py#L132

Reply:
lol. i see. thanks for the explanation